web/massastation/src/pages/Store/Install.tsx

```diff
@@ -80,25 +83,30 @@ function Install({
               <div className="mas-caption text-neutral mb-3">
                 {Intl.t('store.sidepanel.subtitle')}
               </div>
-              {isLoading ? (
-                <div className="mas-body2 mb-4 text-neutral">
-                  {Intl.t('store.loading')}
-                </div>
-              ) : (
-                <>
-                  <Input
-                    placeholder={Intl.t('store.sidepanel.placeholder')}
-                    name="url"
-                    value={url}
-                    onChange={(e) => handleChange(e)}
-                    error={error}
-                    customClass="bg-primary mb-3"
-                  />
+              <>
+                <Input
+                  placeholder={Intl.t('store.sidepanel.placeholder')}
+                  name="url"
+                  value={url}
+                  onChange={handleChange}
+                  error={error}
+                  customClass="bg-primary mb-3"
+                />
+
+                {isLoading ? (
+                  <Button
+                    disabled
+                    preIcon={<Spinner variant="button" />}
+                    customClass="mt-3"
+                  >
+                    {Intl.t('store.sidepanel.button')}
+                  </Button>
+                ) : (
                   <Button type="submit" customClass="mt-3">
                     {Intl.t('store.sidepanel.button')}
                   </Button>
```

Comment:
TBH I am, maybe, realising that we did a mistake in using pre/pos icon.

I guess this would work as expected. BTW it avoids repeat block and will **prove** to us that preIcon/posIcon was not a good idea.
If I am right we would just need to remove the  `align-items: baseline` to have the icons aligned with the button text. Then, if we need pre/pos icons all we need is add before/after the button text. 😜 Isn't it!? 

```suggestion
                  <Button
                    disabled={isLoading}
                  >
                    {isLoading && <Spinner variant="button" />}{Intl.t('store.sidepanel.button')}
                  </Button>
```

Reply:
from my tests it works best with "align-center" ;)

Reply:
see changes to the ui-kit here https://github.com/massalabs/ui-kit/pull/269

Reply:
Approved there!

Reply:
cool ! going to rebase